src/Middleware/Diagnostics.EntityFrameworkCore/test/FunctionalTests/Helpers/StringHelpers.cs

```diff
@@ -12,8 +12,14 @@ public class StringsHelpers
         public static string GetResourceString(string stringName, params object[] parameters)
         {
             var strings = typeof(DatabaseErrorPageMiddleware).GetTypeInfo().Assembly.GetType("Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Strings").GetTypeInfo();
-            var method = strings.GetDeclaredMethods(stringName).Single();
-            return (string)method.Invoke(null, parameters);
+
+            if (parameters.Length > 0)
```

Comment:
I've fixed up the logic here but why is this helper needed in the first place? Why not just use InternalsVisibleTo? This seems unnecessarily hacky.

Reply:
https://github.com/aspnet/AspNetCore/issues/12319 to track, but probably it won't be prioritized.